app/controllers/hq/partners_controller.rb

```diff
@@ -5,4 +5,48 @@ class Hq::PartnersController < ApplicationController
   def index
     @partners= Partner.all
   end
+
+  def edit
+    load_partner
+    load_associations
+  end
+
+  def update
+    load_partner
+    load_associations
```

Comment:
yes I need it in case an edit validation fails.
So to achieve this goal are you talking about creating a new class that references the three objects as part of that single object instantiation?